torch/_functorch/aot_autograd.py

```diff
@@ -1065,7 +1108,41 @@ def aot_dispatch_base(flat_fn, flat_args: List[Tensor], aot_config: AOTConfig):
 
     @wraps(compiled_fw)
     def new_fn(args):
-        fw_outs = call_func_with_args(compiled_fw, args, disable_amp=disable_amp)
+        if _synthetic_base_info is not None:
+            new_inputs, metadata = merge_view_inputs(args, _input_info)
+            args_with_synthetic_bases = new_inputs
+        else:
+            args_with_synthetic_bases = args
+        all_outs = call_func_with_args(compiled_fw, args_with_synthetic_bases, disable_amp=disable_amp)
+
+        assert len(all_outs) == _num_mutated_inputs + _num_outputs
+
+        if _num_mutated_inputs > 0:
+            nonlocal meta
+            assert len(meta.mutated_inp_indices) == _num_mutated_inputs
+            updated_inputs = all_outs[: _num_mutated_inputs]
+            fw_outs = all_outs[_num_mutated_inputs :]
```

Comment:
Python list slicing is an O(n) operation, best to avoid it in the hotpath

Reply:
thanks for the callout - fixed in the updated commit (I have the traced function return a tuple of (updated_inps, fw_outs) so I don't need to bother indexing)

Reply:
bleh - I can't easily do this, because inductor codegen expects that the output will be a flat list of tensors

Reply:
I'm going to just get everything working first, and come back to this / file an issue

Reply:
probably easiest just to manually iterate through the list without doing any slicing (instead of making inductor deal with tuple of list outputs)